### ISSUE TITLE
RayTaskRunner raises serialization error when passing allow_failure wrapped future

### ISSUE THREAD

**Issue by EmilRex** [PrefectHQ/prefect#7590]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

When running a flow using the `RayTaskRunner` where a passed future is wrapped in `allow_failure`, the flow fails with the following error:
```
TypeError: Could not serialize the argument {'result': <prefect.utilities.annotations.allow_failure object at 0x1084c4c40>} for a task or actor prefect.engine.begin_task_run. Check https://docs.ray.io/en/master/ray-core/objects/serialization.html#troubleshooting for more information.
```

### Reproduction

```python3
from prefect import flow, task
from prefect.utilities.annotations import allow_failure
from prefect_ray.task_runners import RayTaskRunner


@task
def task_one():
    return "first_result"


@task
def task_two(result: str):
    pass


@flow(task_runner=RayTaskRunner())
def main():
    first_task = allow_failure(task_one.submit())
    second_task = task_two.submit(first_task)


main()
```

### Error

```python3
12:47:52.382 | INFO    | prefect.engine - Created flow run 'unselfish-tuna' for flow 'main'
12:47:52.385 | INFO    | prefect.task_runner.ray - Creating a local Ray instance
2022-11-18 12:47:54,593	INFO worker.py:1519 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265 
12:47:55.683 | INFO    | prefect.task_runner.ray - Using Ray cluster with 1 nodes.
12:47:55.684 | INFO    | prefect.task_runner.ray - The Ray UI is available at 127.0.0.1:8265
12:47:56.475 | INFO    | Flow run 'unselfish-tuna' - Created task run 'task_one-248cbb82-0' for task 'task_one'
12:47:56.485 | INFO    | Flow run 'unselfish-tuna' - Submitted task run 'task_one-248cbb82-0' for execution.
12:47:56.582 | INFO    | Flow run 'unselfish-tuna' - Created task run 'task_two-983d0d1d-0' for task 'task_two'
12:47:58.446 | ERROR   | Flow run 'unselfish-tuna' - Crash detected! Execution was interrupted by an unexpected exception: Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 436, in ray._raylet.prepare_args_internal
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/_private/serialization.py", line 450, in serialize
    return self._serialize_to_msgpack(value)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/_private/serialization.py", line 428, in _serialize_to_msgpack
    pickle5_serialized_object = self._serialize_to_pickle5(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/_private/serialization.py", line 390, in _serialize_to_pickle5
    raise e
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/_private/serialization.py", line 385, in _serialize_to_pickle5
    inband = pickle.dumps(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 73, in dumps
    cp.dump(obj)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/cloudpickle/cloudpickle_fast.py", line 627, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle '_queue.SimpleQueue' object

The above exception was the direct cause of the following exception:

TypeError: Could not serialize the argument {'result': <prefect.utilities.annotations.allow_failure object at 0x1084c4c40>} for a task or actor prefect.engine.begin_task_run. Check https://docs.ray.io/en/master/ray-core/objects/serialization.html#troubleshooting for more information.

Traceback (most recent call last):
  File "/Users/emilchristensen/github/examples/cuberg4.py", line 22, in <module>
    main()
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/flows.py", line 442, in __call__
    return enter_flow_run_engine_from_flow_call(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 157, in enter_flow_run_engine_from_flow_call
    return anyio.run(begin_run)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/client/utilities.py", line 47, in with_injected_client
    return await fn(*args, **kwargs)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 230, in create_then_begin_flow_run
    state = await begin_flow_run(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 361, in begin_flow_run
    terminal_state = await orchestrate_flow_run(
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py", line 670, in __aexit__
    raise exc_details[1]
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py", line 199, in __aexit__
    await self.gen.athrow(typ, value, traceback)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 1385, in report_flow_run_crashes
    raise exc from None
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 1365, in report_flow_run_crashes
    yield
  File "/opt/homebrew/Cellar/python@3.9/3.9.14/Frameworks/Python.framework/Versions/3.9/lib/python3.9/contextlib.py", line 653, in __aexit__
    cb_suppress = await cb(*exc_details)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 662, in __aexit__
    raise exceptions[0]
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 953, in create_task_run_then_submit
    await submit_task_run(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect/engine.py", line 1014, in submit_task_run
    future = await task_runner.submit(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/prefect_ray/task_runners.py", line 156, in submit
    self._ray_refs[key] = ray_decorator(sync_compatible(call.func)).remote(
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/remote_function.py", line 122, in _remote_proxy
    return self._remote(args=args, kwargs=kwargs, **self._default_options)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/util/tracing/tracing_helper.py", line 307, in _invocation_remote_span
    return method(self, args, kwargs, *_args, **_kwargs)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/remote_function.py", line 402, in _remote
    return invocation(args, kwargs)
  File "/Users/emilchristensen/.virtualenvs/default/lib/python3.9/site-packages/ray/remote_function.py", line 377, in invocation
    object_refs = worker.core_worker.submit_task(
  File "python/ray/_raylet.pyx", line 1787, in ray._raylet.CoreWorker.submit_task
  File "python/ray/_raylet.pyx", line 1791, in ray._raylet.CoreWorker.submit_task
  File "python/ray/_raylet.pyx", line 402, in ray._raylet.prepare_args_and_increment_put_refs
  File "python/ray/_raylet.pyx", line 393, in ray._raylet.prepare_args_and_increment_put_refs
  File "python/ray/_raylet.pyx", line 444, in ray._raylet.prepare_args_internal
TypeError: Could not serialize the argument {'result': <prefect.utilities.annotations.allow_failure object at 0x1084c4c40>} for a task or actor prefect.engine.begin_task_run. Check https://docs.ray.io/en/master/ray-core/objects/serialization.html#troubleshooting for more information.
```

### Versions

```Text
$ prefect version
Version:             2.6.8
API version:         0.8.3
Python version:      3.9.14
Git commit:          68044e28
Built:               Thu, Nov 17, 2022 3:19 PM
OS/Arch:             darwin/arm64
Profile:             emil-default
Server type:         cloud

$ pip freeze | grep prefect-ray
prefect-ray==0.2.1
```

**Comment by madkinsz** [comment]
Work on this happening in #7263

**Comment by madkinsz** [comment]
@ahuang11 is this resolved by #7263 / #8037 ?

**Comment by ahuang11** [comment]
Just tested; works for me!
```
python test.py
10:00:09.612 | INFO    | prefect.engine - Created flow run 'fluffy-pelican' for flow 'main'
10:00:09.618 | INFO    | prefect.task_runner.ray - Creating a local Ray instance
2023-01-09 10:00:13,106	INFO worker.py:1529 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265
10:00:14.478 | INFO    | prefect.task_runner.ray - Using Ray cluster with 1 nodes.
10:00:14.478 | INFO    | prefect.task_runner.ray - The Ray UI is available at 127.0.0.1:8265
10:00:16.463 | INFO    | Flow run 'fluffy-pelican' - Created task run 'task_one-248cbb82-0' for task 'task_one'
10:00:16.491 | INFO    | Flow run 'fluffy-pelican' - Submitted task run 'task_one-248cbb82-0' for execution.
10:00:16.645 | INFO    | Flow run 'fluffy-pelican' - Created task run 'task_two-983d0d1d-0' for task 'task_two'
(begin_task_run pid=12200) 10:00:18.110 | INFO    | Task run 'task_one-248cbb82-0' - Finished in state Completed()
10:00:18.312 | INFO    | Flow run 'fluffy-pelican' - Submitted task run 'task_two-983d0d1d-0' for execution.
(begin_task_run pid=12200) 10:00:18.982 | INFO    | Task run 'task_two-983d0d1d-0' - Finished in state Completed()
10:00:21.789 | INFO    | Flow run 'fluffy-pelican' - Finished in state Completed('All states completed.')
```


```
prefect version
Version:             2.7.7
```
```
prefect_ray.__version__
'0.2.2'
```
